src/Gui/Hooks/useUpdateEffect.luau

```diff
@@ -0,0 +1,18 @@
+-- Inspired by: https://github.com/streamich/react-use/blob/master/src/useUpdateEffect.ts
+
+local Hoarcekat = script:FindFirstAncestor("Hoarcekat")
+local React = require(Hoarcekat.Packages.React)
+
+local useFirstMountState = require(Hoarcekat.Plugin.Gui.Hooks.useFirstMountState)
+
+local function useUpdateEffect(effect, dependencies)
```

Comment:
Are we sure this isn't overkill?

I know `React.useEffect(callback, {})` runs twice in strict mode, but it doesn't look like this should effect us in our use cases of it, which seem to be idempotent.